src/PolygonDraw/PolygonVertex.tsx

```diff
@@ -21,93 +19,99 @@ export interface Props {
     onDragEnd: () => void;
 }
 
-export interface State {
-    isHovered: boolean;
-    isDragged: boolean;
-    latLng: LatLng;
-}
-
-export class PolygonVertex extends React.Component<Props, State> {
-    private circleMarkerElement: CircleMarker;
-    state = {
-        isHovered: false,
-        isDragged: false,
-        latLng: new LatLng(0, 0),
-    };
+export const PolygonVertex: FC<Props> = ({
+    coordinate,
+    isSelected,
+    index,
+    onClick,
+    onDrag,
+    onDragEnd,
+    onDragStart,
+}) => {
+    const [isHovered, setIsHovered] = useState(false);
+    const [isDragged, setIsDragged] = useState(false);
+    const [latLng, setLatLng] = useState(new LatLng(0, 0));
 
-    static getDerivedStateFromProps = (props: Props, state: State) => {
-        return state.isDragged ? state : { ...state, latLng: createLeafletLatLngFromCoordinate(props.coordinate) };
-    };
+    let circleMarkerElement: CircleMarker;
 
-    componentDidMount() {
-        if (this.circleMarkerElement) {
-            this.circleMarkerElement.on('dragstart', this.handleDragStart);
-            this.circleMarkerElement.on('drag', this.handleDrag);
-            this.circleMarkerElement.on('dragend', this.handleDragEnd);
+    useEffect(() => {
+        if (!isDragged && !!coordinate) {
```

Comment:
do we need to check for existance of the `coordinate` here? Props say it is a required property which can only have coordinates object. If I'm not wrong, it means `!!coordinate` is always `true`

Reply:
You are right, the additional check is not needed. Will be removed

Reply:
I've added a fix in this commit: #[fcf9656](https://github.com/freenowtech/react-polygon-editor/pull/27/commits/fcf96560e8d151c524eaf47654efc80e2a7f9a33)